plugins/modules/fusion_ss.py

```diff
@@ -154,6 +153,16 @@ def update_ss(module, fusion, ss):
         )
         patches.append(patch)
 
+    if module.params["hardware_types"] and sorted(
+        module.params["hardware_types"]
+    ) != sorted(ss.hardware_types):
+        patch = purefusion.StorageServicePatch(
+            hardware_types=purefusion.NullableStringArray(
+                module.params["hardware_types"]
```

Comment:
This is a bug in the OpenAPI spec and ability to change hardware types shouldn't be implemented. Backend doesn't support this, it is not possible to actually change anything on the storage service once it is created except the display name at the moment.

Reply:
Hmm, I didn't know it.

Reply:
However, this Ansible Collection should be on parity with the latest Pure Storage Fusion API. Once it is removed from the API, we can remove it from this collection too.

Reply:
Nope, we need to be in parity with backend, that is the source of truth - we should fix the spec